opendut-util/opendut-auth/src/registration/client.rs

```diff
@@ -126,4 +136,92 @@ impl RegistrationClient {
+                 let filtered_clients = clients.0.into_iter().filter(|client| client.base_url.clone().is_some_and(|url| url.contains(&resource_id.value().to_string()))).collect::<Vec<Client>>();
+    
+                 Ok(Clients(filtered_clients))
+            }
+            Err(error) => {
+                 Err(RegistrationClientError::RequestError { error: "OIDC client list request failed!".to_string(), cause: Box::new(error) })
+            }
+        }
+    }
+
+    pub async fn delete_client(&self, resource_id: Id) -> Result<Clients, RegistrationClientError> {
+        let access_token = self.inner.get_token().await
+            .map_err(|error| RegistrationClientError::RequestError { error: error.to_string(), cause: Box::new(error) })?;
+
+        let clients = self.list_clients(resource_id).await?;
+        
+        let mut failed_deletion_clients = Vec::new();
+        
+        for client in clients.clone().0 {
+            let delete_client_url = request_uri(&self.config.issuer_remote_url, Some(client.client_id.to_string()))?;
+
+            let request = create_http_request_client(&access_token, &delete_client_url, http::Method::DELETE)?;
+
+            let response = self.inner.reqwest_client.async_http_client(request)
+                .await;
+
+            if response.is_err() {
+                failed_deletion_clients.push(client.client_id);
+            }
+        }
+        
+        if failed_deletion_clients.is_empty() {
+            Ok(clients)
+        } else {
+            Err( RegistrationClientError::ClientDeletionError { client_ids: failed_deletion_clients.join(",") } )
+        }
+    }
+}
+
+fn request_uri(issuer_remote_url: &Url, client_id: Option<String>) -> Result<Url, RegistrationClientError> {
+    match client_id {
+        Some(client_id) => {
+            issuer_remote_url.join(format!("/admin/realms/opendut/clients/{}", client_id).as_str())
+                .map_err(|error| RegistrationClientError::RequestError { error: error.to_string(), cause: Box::new(error) })
+        }
+        None => {
+            issuer_remote_url.join("/admin/realms/opendut/clients/")
+                .map_err(|error| RegistrationClientError::RequestError { error: error.to_string(), cause: Box::new(error) })
+        }
+    }
+}
+
+fn create_http_request_client(access_token: &Token, issuer_remote_url: &Url, http_method: http::Method) -> Result<HttpRequest, RegistrationClientError> {
+    let mut headers = HeaderMap::new();
+    let bearer_header = format!("Bearer {}", access_token);
+    let access_token_value = HeaderValue::from_str(&bearer_header)
+        .map_err(|error| RegistrationClientError::InvalidConfiguration { error: error.to_string() })?;
+    headers.insert(http::header::AUTHORIZATION, access_token_value);
+
+    Ok(HttpRequest {
+        method: http_method,
+        url: issuer_remote_url.clone(),
+        headers,
+        body: vec![],
+    })
+}
+
```

Comment:
This should be a private method within the registration client. This would make passing the access token obsolete.

Reply:
Done